### PR TITLE
Log gradient norms of any magnitude

### PR DESCRIPTION
**Pull request by martenlienen**
## What does this PR do?

After just having spent 3 hours trying to figure out, why some of parameters apparently receive no gradients, I found out that `grad_norm` arbitrarily decides that gradient norms less than `1e-4` are not worth knowing about and rounds them to zero. This PR fixes that.

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

File: src/lightning/pytorch/utilities/grads.py [hunk 1]
```diff
@@ -48,5 +48,4 @@ def grad_norm(module: Module, norm_type: Union[float, int, str], group_separator
     if norms:
         total_norm = torch.tensor(list(norms.values())).norm(norm_type)
         norms[f"grad_{norm_type}_norm_total"] = total_norm
-    norms = {k: round(v.item(), 4) for k, v in norms.items()}
```
Comment (carmocca) [inline src/lightning/pytorch/utilities/grads.py:51]: For future reference, this was done in the past because this value would be automatically included in the progress bar when `track_gad_norm` was set. So norms this small would clutter the limited space in the progress bar.

This is no longer an issue after https://github.com/Lightning-AI/lightning/pull/16745

**Review by carmocca — APPROVED** [review]
Can you add a CHANGELOG entry? Under the "Changed" section

**Review by awaelchli — APPROVED** [review]
Great!

File: src/lightning/pytorch/utilities/grads.py (hunk 1 above)
Comment (martenlienen) [reply src/lightning/pytorch/utilities/grads.py:51]: I see. I think a better solution would have been to round at the location of display, i.e. before printing to the terminal, instead of during data collection, but now it doesn't matter anymore anyway.

**Comment by martenlienen** [comment]
Thanks for the review and information. I have added a changelog entry.
